wfe2/wfe.go

```diff
@@ -2200,6 +2211,47 @@ func (wfe *WebFrontEndImpl) validateCertificateProfileName(profile string) error
 	return nil
 }
 
+func (wfe *WebFrontEndImpl) checkIdentifiersPaused(ctx context.Context, identifiers []identifier.ACMEIdentifier, uniqueNames []string, regID int64) ([]string, error) {
+	identifierMap := make(map[string]identifier.ACMEIdentifier)
+	for _, ident := range identifiers {
+		identifierMap[ident.Value] = ident
+	}
+
+	uniqueIdentifiers := make([]*sapb.Identifier, 0, len(uniqueNames))
+	for _, name := range uniqueNames {
+		ident, exists := identifierMap[name]
+		if exists {
+			uniqueIdentifiers = append(
+				uniqueIdentifiers,
+				&sapb.Identifier{Type: string(ident.Type), Value: ident.Value},
+			)
+		} else {
+			// This should never happen.
+			return nil, fmt.Errorf("identifier %q not found in identifiers list", name)
+		}
+	}
```

Comment:
This is basically a reimplementation of core.UniqueLowerNames, with two differences:
- the output is a list of identifiers, instead of a list of strings (good)
- it doesn't work if any of the input identifiers are _not_ lowercased, because then the lookup on line 2222 will fail when the identifierMap's key is `Example.Com` but the `name` from `uniqueNames` is `example.com`.

I think this logic should be simplified and moved into `core.UniqueLowerIdentifiers`, and then this function should just call that helper. And we'll slowly migrate the rest of the `UniqueLowerNames` callers to use the new function over time.